File: Learn Java/Baseball Game.java
class Solution {
    public int calPoints(String[] ops) {
        
        Stack<Integer> stk = new Stack<>();
        for (int i=0;i<ops.length;i++){
            if (ops[i].equals("C")){
                stk.pop();
            }
            else if (ops[i].equals("D")){
                int d = 2* stk.peek();
                stk.push(d);
            }
            else if (ops[i].equals("+")){
                int s = 0;
                int f = stk.peek();
                s+=f;
                stk.pop();
                s += stk.peek();
                stk.push(f);
                stk.push(s);
            }
            else{
                int a = Integer.parseInt(ops[i]);
                stk.push(a);
            }
        }
        
        int sum = 0;
        for(int val : stk) sum+= val;
        return sum;
    }
}

File: Data Structure/1.Array/baseball_game.py
# https://leetcode.com/problems/baseball-game/
class Solution:
    def calPoints(self, ops: List[str]) -> int:
        stack = []
        for i in ops:
            if i=='C':
                stack.pop()
            elif i=='D':
                el = stack[-1]
                stack.append(el*2)
            elif i=='+':
                last = stack[-1]
                secLast = stack[-2]
                stack.append(last+secLast)
                
            else:
                stack.append(int(i))
                
        return sum(stack)
        